raiden_contracts/contracts/TokenNetwork.sol

```diff
@@ -544,6 +548,20 @@ contract TokenNetwork is Utils {
         private
         returns (uint256, uint256, uint256, uint256)
     {
+        // Cases require attention:
+        // case1. If participant1 does NOT provide a balance proof or provides an old balance proof.
+        // participant2_transferred_amount can be [0, real_participant2_transferred_amount)
+        // We need to punish participant1.
+        // case2. If participant2 does NOT provide a balance proof or provides an old balance proof.
+        // participant1_transferred_amount can be [0, real_participant1_transferred_amount)
+        // We need to punish participant2.
+        // case3. If neither participants provide a balance proof, we just subtract their
+        // withdrawn amounts from their deposits.
+
+        // case1 is solved by making sure we only have to handle case2 or case3:
+        require(participant2_transferred_amount >= participant1_transferred_amount);
```

Comment:
This is already done in line 460

Reply:
I really want this check to stay with the `getSettleTransferAmounts` function, to protect from people changing the code without knowing the importance of it.

I would rather remove the one from 460. 
But I also wanted:
- the main `settleChannel` to be aware that the order is important (I can just add a comment in the docstring to solve this one)
- fail faster and return more gas to the sender. (not really an issue)

So, I am fine with adding the info the docstring + removing line 460.

Reply:
You could add a comment in order _to protect from people changing the code without knowing the importance of it_ and get the other two for free.